### PR TITLE
Check Markdown with Markdownlint

### PR DESCRIPTION
**Pull request by jlibovicky**
And now a controversial one: Should we become style nazis and check markdown with markdownlint?

This PR introduces markdownlint into lint tests suite.

### CODE REVIEW

File: tests/lint_run.sh [hunk 1]
```diff
@@ -1,3 +1,7 @@
 #!/bin/bash
 
+set -e
+
 pylint -j 4 $(grep --include='*.py' --exclude='__init__.py' -rl neuralmonkey -e "^# *tests:.*lint")
+
+mdl $(find neuralmonkey -name '*.md') $(find tests -name '*.md') $(find examples -name '*.md')
```
Comment (tomasmcz) [inline tests/lint_run.sh]: What about `README.md` in the root directory?

**Comment by tomasmcz** [comment]
I'm all for automatic checking of everything that's possible.

Maybe this should be a separate test suite, so pylint can be run separately, without installing and running mdl.

**Comment by jindrahelcl** [comment]
Would you please write a short description of what this module checks and put it either to readme or to a separate page in documentation?

**Comment by jindrahelcl** [comment]
Oh one more thing - the docs are written in reStructuredText - does this tool check this also?

**Comment by jlibovicky** [comment]
It does not check rst. I though everything was  in markdown for some reason. Shouldn't unite all the documentation format then?

**Comment by jindrahelcl** [comment]
The thing is, the documentation uses Sphinx, which afaik does not support markdown, but only rst (that is, docstrings should use rst). I also think rst is more expressive than markdown, but I didn't actually did a proper research on this. The rst alternative seemed easier to automatize.

**Comment by jindrahelcl** [comment]
rst linter is also available: https://pypi.python.org/pypi/restructuredtext_lint
Considering the amount of text written in each of these, I'd prefer to switch to rst, in case we decide to unite the styles. I don't like the format better, though..

**Comment by jindrahelcl** [comment]
Having read this article, http://zverovich.net/2016/06/16/rst-vs-markdown.html, I'd stick with rst, despite it being more complicated.

**Comment by jlibovicky** [comment]
It's possible to use Markdown with Sphinx: https://github.com/serra/sphinx-with-markdown/blob/master/docs/conf.py#L342
And existing documentation can be converted to Markdown using pandoc, hm?

**Comment by tomasmcz** [comment]
Markdown is the _de facto_ Github documentation standard, RST is the _de facto_ Python documentation standard, so it makes sense to use each one for its purpose. On the other hand, having just one format might be easier to maintain. Github seems to handle RST and it would certainly be easier to convert the few markdown files than the other way round. On the other hand, I like the markdown syntax better. If I had to choose, I would reluctantly go with markdown everywhere.

**Comment by jindrahelcl** [comment]
The syntax is not *that* worse.. If you want to write a plain document as readme, the source will be almost the same. When I was writing the tutorial in rst, it was ok.

**Comment by jindrahelcl** [comment]
Also I agree that it's ok to use each one for its purpose, e.g. md for github things (of which there is not going to be many) and rst for documentation and docstrings. (Most of the docstrings are compatible with both, anyway.) I cannot imagine a situation when there is a problem of having both, provided that the file names have the proper extensions. After all, we do have both now, and there is no problem, so I wouldn't solve anything that does not need solving, I'd just add the rst linter to the test suite.

**Comment by jlibovicky** [comment]
OK, let's have dual have rst for Python and md for github. I would however merge this PR before someone change the markdown files. The rst linter complains about some stuff I don't want to deal with now. Moreover, we should later check also the docstrings using the rst linter.

**Pull request merged**